chess.js

```diff
@@ -225,6 +228,55 @@
       return false;
     }
 
+    if (turn == 0 && white_can_castle){ // white castle
+      if(a1 == 7 && a2 == 4){
+        if(b1 == 7 && b2 == 6 && // white kingside castle
+          board[7][7] == 'R' && board[7][6] == '' && board[7][5] == ''){
```

Comment:
@chezbgone  Hi, I believe you need to check that the king is in the square, also that the side is white.